apollo-router/src/spec/query/transform.rs

```diff
@@ -827,5 +853,85 @@ fragment F on Query {
         let doc = ast::Document::parse(query, "query.graphql").unwrap();
         let result = document(&mut visitor, &doc).unwrap();
         insta::assert_snapshot!(TestResult { query, result });
+
+        // test removed field with variable in list argument
+        let query = r#"
+            query($a: String, $b: String) {
+                c
+                d(arg: ["a", $a, "b"]) @remove
```

Comment:
i am going to be very pedantic 😅😅😅. 

can these tests also test for a variable in list in list, and a variable in object in a list. example (and not a semantically correct one, so just to illustrate a point):
* d(arg: ["a", [$a], "b"]) @remove
* d(arg: ["a", {c: $a}, "b"]) @remove

Reply:
sure thing: [ce12831](https://github.com/apollographql/router/pull/6025/commits/ce128316ff95394bb1a0017bd42b6a89990ee844)